openstf/exceptions.py

```diff
@@ -41,3 +42,19 @@ def __init__(
         self.pid = pid
         self.message = message
         super().__init__(self.message)
+
+
+class InputDataInvalidError(Exception):
+    """Invalid input data"""
+
+
+class InputDataInsufficientError(InputDataInvalidError):
+    """Insufficient input data"""
+
+
+class InputDataWrongColumnOrderError(InputDataInvalidError):
+    """Wrong column order input data"""
+
+
+class OldModelHigherScoreError(Exception):
+    """Old model has a higher score then new model"""
```

Comment:
Suggestion, can we include a `pid` argument with this exception definition? That would make it more easy to see for which pid the exception was raised.

Reply:
Good point, I agree that you want to know for which `pid` a certain exception is raised. I'm not sure if this knowledge belongs to the exception itself. I would suggest to add it to the logging (if not already done so) or to the exception message when we raise the exception.

update, I just checked. In case of the `train_model_pipeline()` and `train_model_pipeline_core()` function we use the prediction job name in the exception: 
![image](https://user-images.githubusercontent.com/6715707/130236109-c97294c5-b2d8-4437-b184-3498e0ff2075.png)
![image](https://user-images.githubusercontent.com/6715707/130236244-733d269d-f9d7-46c4-8fcd-980999a32994.png)

We could change that to the prediction job id, perhaps that makes more sense?

Reply:
Putting it in the logging is also a valid option. Is remembered seeing the inclusing in the exception itself somewhere else in openstf: 
https://github.com/alliander-opensource/openstf/blob/main/openstf/exceptions.py

I guess either the pj['name']  or id would work, but perhaps nice to pick 1. Feel free to pick;)